plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts

```diff
@@ -574,25 +412,104 @@ export async function readMicrosoftGraphOrg(
       transformer: options.userTransformer,
       logger: options.logger,
     });
-    users.push(...usersWithFilter);
+    users = usersWithFilter;
+  } else {
+    options.logger?.info(
+      'Not loading users as neither user or userGroupMember configuration has been provided',
+    );
+    users = [];
   }
-  const { groups, rootGroup, groupMember, groupMemberOf } =
-    await readMicrosoftGraphGroups(client, tenantId, {
+
+  let groups: GroupEntity[];
+  if (options.groupSearch || options.groupFilter) {
+    const {
+      groups: groups2,
+      rootGroup,
+      groupMember,
+      groupMemberOf,
+    } = await readMicrosoftGraphGroups(client, tenantId, {
       queryMode: options.queryMode,
       groupSearch: options.groupSearch,
       groupFilter: options.groupFilter,
       groupSelect: options.groupSelect,
       groupTransformer: options.groupTransformer,
       organizationTransformer: options.organizationTransformer,
+      logger: options.logger,
     });
 
-  resolveRelations(rootGroup, groups, users, groupMember, groupMemberOf);
+    groups = groups2;
+
+    options.logger?.info('Resolving relations');
+    resolveRelations(rootGroup, groups, users, groupMember, groupMemberOf);
+  } else {
+    options.logger?.info(
+      'Not loading groups as no group configuration has been provided',
+    );
+    groups = [];
+  }
+
   users.sort((a, b) => a.metadata.name.localeCompare(b.metadata.name));
   groups.sort((a, b) => a.metadata.name.localeCompare(b.metadata.name));
 
+  if (users.length === 0 && groups.length === 0) {
+    options.logger?.warn(
+      'Microsoft Graph Entity Provider loaded no users or groups. Check your configuration.',
```

Comment:
So this will warn even if you explicitly asked to not get any users or groups?

Reply:
My thinking here is that the MS graph plugin isn't installed by default.  So it seems very odd that you'd go to all the effort of setting the provider up to not have it ingest anything.  It was also partly there as a catch all for someone who had no config prior to the upgrade, that is now getting 0 users and groups.  There should be seperate logs showing 0 users and 0 groups were ingested so you wouldn't loose information.  I just through it was worth a warnign specifically to highlight.

I don't feel too strongly though, so happy to remove if you're not convinced it'd useful.